pipelines/pipelines.go

```diff
@@ -525,3 +517,69 @@ func Reduce[S, T string](ctx context.Context, in <-chan S, f func(T, S) T) (T, e
 		}
 	}
 }
+
+// ErrorSink provides an error-handling solution for pipelines created by this package. It manages a
+// pipeline stage which can receive fatal and non-fatal errors that may occur during the course of a pipeline.
+type ErrorSink struct {
+	errors chan errWrapper
+	cancel context.CancelFunc
+	errs   []error
+}
+
+// NewErrorSink returns a new ErrorSink, along with a context which is cancelled when a fatal error is sent to the
+// ErrorSink. Starts a new, configurable pipeline stage which catches any errors reported.
+func NewErrorSink(ctx context.Context, opts ...Option) (context.Context, *ErrorSink) {
+	ctx, cancel := context.WithCancel(ctx)
+	result := &ErrorSink{cancel: cancel}
+	doWithConf(ctx, func(ctx context.Context, in ...chan errWrapper) {
+		result.doErrSink(ctx, in[0])
+	}, configure(opts))
+	return ctx, result
+}
+
+func (s *ErrorSink) doErrSink(ctx context.Context, errors chan errWrapper) {
+	s.errors = errors
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case werr := <-errors:
+			s.errs = append(s.errs, werr.err)
```

Comment:
There's a race condition here when WithPool is used.

Reply:
fixed.